dashboard/app/controllers/regional_partners_controller.rb

```diff
@@ -0,0 +1,79 @@
+class RegionalPartnersController < ApplicationController
+  authorize_resource class: :pd_regional_partner_management
+  before_action :set_regional_partner, only: [:show, :edit, :update]
```

Comment:
Instead of doing this manually, leverage CanCan's `load_resource`. Even better, combine this with the previous line and use [load_and_authorize_resource](https://github.com/CanCanCommunity/cancancan/wiki/authorizing-controller-actions)